### ISSUE TITLE
error CS0122: 'CodeFunctionNode' is inaccessible due to its protection level

### ISSUE THREAD

**Issue by nukadelic** [gilescoope/shader-graph-nodes#3]
The code looks very good, was very hyped to try it but I'm primarily using unity version above 2019.1 and it seems like shader graph classes are now internally implemented is internal and no longer work. 

https://forum.unity.com/threads/unable-to-create-custom-shader-graph-node-due-to-inaccessibility-of-customfunctionnode-class.586876/

But there is hope for the upcoming version 7.0.0 
https://github.com/Unity-Technologies/ScriptableRenderPipeline/blob/master/com.unity.shadergraph/CHANGELOG.md

**Comment by gilescoope** [comment]
Yeah unfortunately Unity removed this method of generating shader nodes. There is now a custom function node where custom shaders can be typed in. It isn't as robust however as the old method. In any case you should be able to copy the code for any of the shader nodes I've written here and set up the inputs and outputs correctly yourself.

**Comment by CConchelos** [comment]
Hey there sir, I would like to use your SDF shaders. Could you take one - say circle - and give me an example of how to port it to a custom node? If I had a template to follow, then I could convert the rest of them. I was super jazzed to try out these shaders tonight, but I can see that I don't know enough to port this.

**Comment by neon-age** [comment]
I've added support for newer Unity versions by using AsmRef! See #5 🙂
